hub/core/tensor.py

```diff
@@ -108,7 +110,18 @@ def extend(self, samples: Union[np.ndarray, Sequence[SampleValue]]):
         Args:
             samples (np.ndarray, Sequence, Sequence[Sample]): The data to add to the tensor.
                 The length should be equal to the number of samples to add.
+
+        Raises:
+            TensorDtypeMismatchError: TensorDtypeMismatchError: Dtype for array must be equal to or castable to this tensor's dtype
         """
+        if self.meta.dtype:
+            err_dtype = get_incompatible_dtype(samples, self.meta.dtype)
+            if err_dtype:
+                raise TensorDtypeMismatchError(
+                    self.meta.dtype,
+                    err_dtype,
+                    self.meta.htype,
+                )
```

Comment:
we should probably remove this (or move it into the `intelligent_cast` function)

Reply:
no need to check `if self.meta.dtype`